raft/raft.go

```diff
@@ -278,7 +280,15 @@ func (r *raft) sendAppend(to uint64) {
 			}
 		}
 	}
-	r.send(m)
+	// If there is a race between the checking of the need of a snapshot and
+	// an external async log compaction, the log term in the msgApp might
+	// be maxUint64 or the err is not nil (when there is an ErrCompact returned from storage)
+	// Retry sendAppend to resolve this low probability issue.
+	if m.Type == pb.MsgApp && (m.LogTerm == math.MaxUint64 || err != nil) {
```

Comment:
Again, this should probably be `err == ErrCompacted` instead of `err != nil` (with a panic for other errors).

Would it be better to add an `error` return to `raftLog.term()` instead of using `MaxUint64` as a special value? This way we don't have to worry about other places that call term() failing to check for this case.